### PR TITLE
Update terraform for eu region support

### PR DESCRIPTION
**Pull request by Lfdelossantos**
PR for ticket: https://segment.atlassian.net/browse/DL-573

Currently, terraform-segment-data-lakes repo supports creating DL infra for US DLs. The same TF would not work for EU DLs because the IAM roles that EU uses to assume customers' roles is different. This is a short term solution that accepts the region as a config (default to US) and based on that, add the appropriate roles to the trust relationship.

This is done like this:
Use the preexisting config data to find the aws account region
Then used a conditional to set the ARN based on the region value. 

Testing:

Created Main.tf for sauron-euw workspace with aws_region as us-west-2 and segment_region eu-west-1

this is the aws_region which worked correctly
image

This is the result of the conditional which has the correct ARN for EUW
image

After applying the plan and checking IAM policies in AWS dev-read this is the result 
image

Now to check the US behavior (default) 
Created a new main.tf for the US sauron workspace with aws_region and segment_region as us-west-2

This is the aws_region which worked correctly
image

This is the iam roles which worked correctly
image

After running terraform apply and checking IAM on AWS console this is the result 
image

### CODE REVIEW

File: aws_datalake/modules/iam/main.tf [hunk 1]
```diff
@@ -7,6 +7,9 @@ resource "aws_iam_role" "segment_data_lake_iam_role" {
   tags               = local.tags
 }
 
+# https://www.terraform.io/docs/providers/aws/d/region.html
+data "aws_region" "current" {}
```
Comment (uditmehta27) [inline aws_datalake/modules/iam/main.tf:11]: This is the region where the DL is being set up right? Customers can use Segment in the US but their DL can be configured in EU. We need this to be a different config than the region the customer is setting up their DL in. Something like `segment_region` or on those lines.

File: aws_datalake/README.md [hunk 2]
```diff
@@ -88,22 +88,25 @@ resource "aws_s3_bucket" "segment_datalake_s3" {
...
+  source = "git@github.com:segmentio/terraform-segment-data-lakes//aws_datalake/modules/iam?ref=v0.8.0"
 
   # Suffix is not strictly required if only initializing this module once.
   # However, if you need to initialize multiple times across different Terraform
   # workspaces, this hook allows the generated IAM policies to be given unique
   # names.
   suffix = "-prod"
-
+  
+  # segment_region is an optional field which helps you to specify the region for your datalake. The default is us_west_2
```
Comment (uditmehta27) [inline aws_datalake/README.md:99]: I would rename this to "...specify the region if you have configured Segment in a different region. See https://segment.com/docs/guides/regional-segment/ for more details". Or something on those lines to make it clear.

**Review by uditmehta27 — APPROVED** [review]
LGTM! Small comment to update the documentation.

**Comment by Lfdelossantos** [comment]
atlantis apply

**Pull request merged**